microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/config/java/MicroProfileConfigASTValidator.java

```diff
@@ -187,7 +189,11 @@ private void validatePropertyHasValue(Annotation annotation, Expression defaultV
 				name = MicroProfileConfigPropertyProvider.getPropertyName(name, currentPrefix);
 			}
 
-			if (name != null && !hasDefaultValue && !doesPropertyHaveValue(name, getContext())
+			if (name != null && name.isEmpty()) {

```

Comment:
I will write just one if (name != null) and after that write if (name.isEmpty()))